File: mobile/src/main/java/technology/mainthread/apps/gatekeeper/model/event/StringDefUtil.java
package technology.mainthread.apps.gatekeeper.model.event;

import android.support.annotation.NonNull;

public class StringDefUtil {

    private StringDefUtil() {
    }

    public static @NonNull @GatekeeperState String getGatekeeperState(String state) {
        if (state != null) {
            String gatekeeperState = state.toUpperCase();
            switch (gatekeeperState) {
                case GatekeeperState.OFFLINE:
                    return GatekeeperState.OFFLINE;
                case GatekeeperState.ONLINE:
                    return GatekeeperState.ONLINE;
                case GatekeeperState.DOOR_OPEN:
                    return GatekeeperState.DOOR_OPEN;
                case GatekeeperState.PRIMED:
                    return GatekeeperState.PRIMED;
                default:
                    break;
            }
        }
        return GatekeeperState.UNKNOWN;
    }

}

File: mobile/src/main/java/technology/mainthread/apps/gatekeeper/model/event/StringDefUtil.kt
package technology.mainthread.apps.gatekeeper.model.event

fun getGatekeeperState(state: String?): GatekeeperState {
    if (state != null) {
        val gatekeeperState = state.toUpperCase()
        when (gatekeeperState) {
            GatekeeperState.OFFLINE.label -> return GatekeeperState.OFFLINE
            GatekeeperState.ONLINE.label -> return GatekeeperState.ONLINE
            GatekeeperState.DOOR_OPEN.label -> return GatekeeperState.DOOR_OPEN
            GatekeeperState.PRIMED.label -> return GatekeeperState.PRIMED
            else -> {
            }
        }
    }
    return GatekeeperState.UNKNOWN
}
